configure.ac

```diff
@@ -546,7 +546,28 @@ esac
 
 AC_ARG_WITH(iodbc,
 AS_HELP_STRING([--with-iodbc=DIR], [build odbc driver against iODBC in DIR]))
-if test "x$with_iodbc" != "x" -a "x$with_iodbc" != "xno"; then
+
+if test "x$with_iodbc" = "xyes"; then
+	# The user asked for iodbc support, but didn't tell us
+	# where to look. So, we'll try to get the info from
+	# pkg-config. If we can't, we'll display an error.
+	PKG_CHECK_MODULES(IODBC, libiodbc)
+	CPPFLAGS="$CPPFLAGS -DIODBC $IODBC_CFLAGS"
+	LDFLAGS="$LDFLAGS $IODBC_LIBS"
```

Comment:
This is what the comment is talking about.. neither choice is really good. The `IODBC_LIBS` variable can contain both the `-lfoo` and `-L/usr/foo` type of linker flags.

By placing `IODBC_LIBS` in `LDFLAGS`, we pass some extra `-l` flags to every linker command.

On the other hand, the `ODBCLIB` variable appears in some places where `-L/usr/foo` flags would cause problems:

* `bsqlodbc_LDADD	= $(ODBCLIB) $(ODBCNODMLIB)` in src/apps/Makefile.am
* `LIBS		=	$(ODBCLIB)...` in src/odbc/unittests/Makefile.am

What we really want is a set of LDFLAGS that only applies to the ODBC stuff, so that's why I mentioned a potential refactoring to involve an `ODBC_LDFLAGS` variable.